Services/AspireService.cs

```diff
@@ -1,93 +1,119 @@
+            var pod = podsList.Items[0];
+            var initialData = new WatchResourcesUpdate()
+            {
+                InitialData = new InitialResourceData()
+                {
+                    ResourceTypes = { new ResourceType() { UniqueName = "Pod" } },
+                    Resources = { new Resource() { Name = "Pod", Endpoints = { new Aspire.V1.Endpoint() { EndpointUrl = pod.Status.PodIP } }, Commands = { new ResourceCommandRequest() { CommandType = "Restart" } } }
+
+                    }
+                }
+            };
+            return initialData;           
+        }
+
+        private async Task WatchAndWriteChangesToStream(IServerStreamWriter<WatchResourcesUpdate> responseStream)
+        {
+            // Creates the watcher
+            var podsWatchResponse = await _kubernetesClient.CoreV1.ListNamespacedPodWithHttpMessagesAsync(_kubernetesConfig.Namespace, watch: true);
+            var podWatcher = podsWatchResponse.Watch<V1Pod, V1PodList>(async (type, item) =>
+            {
+                
+                Console.WriteLine($"Pod event of type {type} detected for {item.Metadata.Name}");
+                var reply = new WatchResourcesUpdate()
+                {
+                    // TODO : Figure out deletion or upsert and add here.
+                    Changes = new WatchResourcesChanges()
+                    {
+                        Value = { new WatchResourcesChange() { Delete = new ResourceDeletion() { }, Upsert = new Resource { Name = item.Metadata.Name} } }
+                    }
+                    
+                };
+                await responseStream.WriteAsync(reply);
+            });
+        }
+
+        public override async Task WatchResourceConsoleLogs(WatchResourceConsoleLogsRequest request, IServerStreamWriter<WatchResourceConsoleLogsUpdate> responseStream, ServerCallContext context)
+        {
+           var label = $"app={request.ResourceName}";
+            _logger.LogInformation($"Label : {label}");
+           var pods = await _kubernetesClient.CoreV1.ListNamespacedPodAsync(_kubernetesConfig.Namespace, labelSelector: label);
+           var podName = pods.Items[0].Metadata.Name;
+            _logger.LogInformation($"PodName: {podName}");
+           while(!context.CancellationToken.IsCancellationRequested)
+            {
+                var stream = await _kubernetesClient.CoreV1.ReadNamespacedPodLogAsync(podName, _kubernetesConfig.Namespace);
+                var logsUpdate = new WatchResourceConsoleLogsUpdate();
+                using var reader = new StreamReader(stream);
+                while (!reader.EndOfStream)
+                {
+                    var logEntry = await reader.ReadLineAsync();
+                    logsUpdate.LogLines.Add(new ConsoleLogLine { Text = logEntry });
+                }
+                await responseStream.WriteAsync(logsUpdate);
+                // Introduce delay for testing

```

Comment:
Nit: is this only for testing? It makes sense to me to have some delay there in all cases. Else we would send requests to the Kubernetes API server all the time.